src/shared/reboot-util.c

```diff
@@ -107,3 +125,63 @@ int shall_restore_state(void) {
 
         return r > 0 ? ret : true;
 }
+
+static int xen_kexec_loaded(void) {
+#if HAVE_XENCTRL
+        size_t size;
+        _cleanup_close_ int privcmd_fd = -1;
+        _cleanup_close_ int buf_fd = -1;
+        void *buffer;
+        xen_kexec_status_t *status;
+        int r;
+
+        size = page_size();;
+        if (sizeof(*status) > size)
```

Comment:
this cannot really happen, can it? this would be a programming error, not a realistic error no? hence maybe make this an assert()?

Reply:
I doubt we'd ever see a system where a page size would be less than the kexec request structure, but I'd still rather see a nice error in case page_size() misbehaves, than an assert.